models/migrations/v1_22/v284.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+package v1_22 //nolint
+
+import (
+	"errors"
+	"fmt"
+
+	"code.gitea.io/gitea/modules/log"
+	"code.gitea.io/gitea/modules/setting"
+
+	"xorm.io/xorm"
+)
+
+func ExpandHashReferencesToSha256(x *xorm.Engine) error {
```

Comment:
I wonder - if Sync() do convert var-char length automatically?

Reply:
I think I've tried that in the past and it failed.  Maybe it wasn't possible to change types with Sync() in the past as there is also similar approach to mine in older migrations (ie. my inspiration!). But I'm definitely not expert here. It is possible I've made a mistake in my approach.